cibuildwheel/macos.py

```diff
@@ -579,17 +579,41 @@ def build(options: Options, tmp_path: Path) -> None:
                         shell_with_arch(before_test_prepared, env=virtualenv_env)
 
                     # install the wheel
+                    if (
+                        sys.version_info <= (3, 8)
+                        and testing_arch == "x86_64"
+                        and python_arch == "x86_64"
+                    ):
```

Comment:
```suggestion
                    if is_cp38 and python_arch == "x86_64":
```

sys.version_info is the wrong Python version, we want the builder version, not the host

I also don't think we should do the workaround for versions other than 3.8, as users of these wheels probably won't be able to install them (without setting SYSTEM_VERSION_COMPAT themselves). Wheel maintainers should be made aware of this.

Additionally, `testing_arch == "x86_64"` is redundant when `python_arch == "x86_64"`.